homeassistant/components/websocket_api/messages.py

```diff
@@ -65,12 +65,27 @@ def construct_result_message(iden: int, payload: str) -> str:
     return f'{{"id":{iden},"type":"result","success":true,"result":{payload}}}'
 
 
-def error_message(iden: int | None, code: str, message: str) -> dict[str, Any]:
+def error_message(
+    iden: int | None,
+    code: str,
+    message: str,
+    translation_key: str | None = None,
+    translation_domain: str | None = None,
+    translation_placeholders: dict[str, Any] | None = None,
+) -> dict[str, Any]:
     """Return an error result message."""
+    error_payload: dict[str, Any] = {
+        "code": code,
+        "message": message,
+    }
+    if translation_key is not None:
+        error_payload["translation_key"] = translation_key
+        error_payload["translation_placeholders"] = translation_placeholders or {}
+        error_payload["translation_domain"] = translation_domain or const.DOMAIN
```

Comment:
Why do we need these default values?

Reply:
If translation domain is not set (e.g. for `ServiceNotFound`) it defaults to `websocket_api`. I do not want to import the domain const to the exception. Do you think I should do that?
Placeholders are added if the `translation_key` is set, may be this is not strictly needed.

Reply:
Added a comment to explain why a default for `translation_domain` was added.

Reply:
I don't think it's a good solution to have to copy translations of exceptions to the `websocket_api` integration.
Either we give some special meaning to a `None` `translation_key` so frontend knows to pick them from `common::exceptions`, or we move the standard exception translations to the `homeassisant` integration.